parsl/executors/globus_compute.py

```
@@ -0,0 +1,171 @@
+            one will be instantiated.
+
+        resource_specification:
+            Specify resource requirements for individual task execution.
+
+        user_endpoint_config:
+            User endpoint configuration values as described
+            and allowed by endpoint administrators. Must be a JSON-serializable dict
+            or None. Refer docs from `globus-compute
+            <https://globus-compute.readthedocs.io/en/latest/endpoints/endpoints.html#templating-endpoint-configuration>`_
+            for more info.
+
+        label:
+            a label to name the executor
+
+        batch_size:
+            the maximum number of tasks to coalesce before
+            sending upstream [min: 1, default: 128]
+
+        amqp_port:
+            Port to use when connecting to results queue. Note that the
+            Compute web services only support 5671, 5672, and 443.
+
+        client:
+            instance of globus_compute_sdk.Client to be used by the executor.
+            If not provided, the executor will instantiate one with default arguments.
+
+        kwargs:
+            Other kwargs listed will be passed through to globus_compute_sdk.Executor
+            as is. Refer to `globus-compute docs
+            <https://globus-compute.readthedocs.io/en/latest/reference/executor.html#globus-compute-executor>`_
+        """
+        super().__init__()
+        self.endpoint_id = endpoint_id
+        self.task_group_id = task_group_id
+        self.resource_specification = resource_specification
+        self.user_endpoint_config = user_endpoint_config
+        self.label = label
+        self.batch_size = batch_size
+        self.amqp_port = amqp_port
+        self.client = client
+        self.executor_kwargs = kwargs
+
+        if not _globus_compute_enabled:
+            raise OptionalModuleMissing(
+                ['globus-compute-sdk'],
+                "GlobusComputeExecutor requires globus-compute-sdk installed"
+            )
```

To elide useless work in the `_globus_compute_enabled is False` case, consider making this the very first statement in `__init__()`.  No need to do the the other work if it's going to be thrown away anyway.  (I get that this change is minimal, but it's the principal of fail early and fail fast that I'm after.)